<!DOCTYPE html>
<html lang="en">

<head>
  <script src="/gdpr/gdprscript.js?buildTime=1234&hasRemindMe=true"></script>
  <title>Competition tasks - IDASH PRIVACY & SECURITY WORKSHOP 2022 - secure genome analysis competition</title>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <link id="wsite-base-style" rel="stylesheet" type="text/css"
    href="http://cdn2.editmysite.com/css/sites.css?buildTime=1234" />
  <link rel="stylesheet" type="text/css" href="http://cdn2.editmysite.com/css/old/fancybox.css?1234" />
  <link rel="stylesheet" type="text/css" href="files/main_style.css?1528327367" title="wsite-theme-css" />
  <link href='http://fonts.googleapis.com/css?family=Actor&subset=latin,latin-ext' rel='stylesheet' type='text/css' />
  <style type='text/css'>
    .wsite-elements.wsite-not-footer:not(.wsite-header-elements) div.paragraph,
    .wsite-elements.wsite-not-footer:not(.wsite-header-elements) p,
    .wsite-elements.wsite-not-footer:not(.wsite-header-elements) .product-block .product-title,
    .wsite-elements.wsite-not-footer:not(.wsite-header-elements) .product-description,
    .wsite-elements.wsite-not-footer:not(.wsite-header-elements) .wsite-form-field label,
    .wsite-elements.wsite-not-footer:not(.wsite-header-elements) .wsite-form-field label,
    #wsite-content div.paragraph,
    #wsite-content p,
    #wsite-content .product-block .product-title,
    #wsite-content .product-description,
    #wsite-content .wsite-form-field label,
    #wsite-content .wsite-form-field label,
    .blog-sidebar div.paragraph,
    .blog-sidebar p,
    .blog-sidebar .wsite-form-field label,
    .blog-sidebar .wsite-form-field label {}

    #wsite-content div.paragraph,
    #wsite-content p,
    #wsite-content .product-block .product-title,
    #wsite-content .product-description,
    #wsite-content .wsite-form-field label,
    #wsite-content .wsite-form-field label,
    .blog-sidebar div.paragraph,
    .blog-sidebar p,
    .blog-sidebar .wsite-form-field label,
    .blog-sidebar .wsite-form-field label {}

    .wsite-elements.wsite-footer div.paragraph,
    .wsite-elements.wsite-footer p,
    .wsite-elements.wsite-footer .product-block .product-title,
    .wsite-elements.wsite-footer .product-description,
    .wsite-elements.wsite-footer .wsite-form-field label,
    .wsite-elements.wsite-footer .wsite-form-field label {}

    .wsite-elements.wsite-not-footer:not(.wsite-header-elements) h2,
    .wsite-elements.wsite-not-footer:not(.wsite-header-elements) .product-long .product-title,
    .wsite-elements.wsite-not-footer:not(.wsite-header-elements) .product-large .product-title,
    .wsite-elements.wsite-not-footer:not(.wsite-header-elements) .product-small .product-title,
    #wsite-content h2,
    #wsite-content .product-long .product-title,
    #wsite-content .product-large .product-title,
    #wsite-content .product-small .product-title,
    .blog-sidebar h2 {}

    #wsite-content h2,
    #wsite-content .product-long .product-title,
    #wsite-content .product-large .product-title,
    #wsite-content .product-small .product-title,
    .blog-sidebar h2 {}

    .wsite-elements.wsite-footer h2,
    .wsite-elements.wsite-footer .product-long .product-title,
    .wsite-elements.wsite-footer .product-large .product-title,
    .wsite-elements.wsite-footer .product-small .product-title {}

    #wsite-title {}

    .wsite-menu-default a {}

    .wsite-menu a {}

    .wsite-image div,
    .wsite-caption {}

    .galleryCaptionInnerText {}

    .fancybox-title {}

    .wslide-caption-text {}

    .wsite-phone {}

    .wsite-headline,
    .wsite-header-section .wsite-content-title {}

    .wsite-headline-paragraph,
    .wsite-header-section .paragraph {}

    .wsite-button-inner {}

    .wsite-not-footer blockquote {}

    .wsite-footer blockquote {}

    .blog-header h2 a {}

    #wsite-content h2.wsite-product-title {}

    .wsite-product .wsite-product-price a {}

    @media screen and (min-width: 767px) {

      .wsite-elements.wsite-not-footer:not(.wsite-header-elements) div.paragraph,
      .wsite-elements.wsite-not-footer:not(.wsite-header-elements) p,
      .wsite-elements.wsite-not-footer:not(.wsite-header-elements) .product-block .product-title,
      .wsite-elements.wsite-not-footer:not(.wsite-header-elements) .product-description,
      .wsite-elements.wsite-not-footer:not(.wsite-header-elements) .wsite-form-field label,
      .wsite-elements.wsite-not-footer:not(.wsite-header-elements) .wsite-form-field label,
      #wsite-content div.paragraph,
      #wsite-content p,
      #wsite-content .product-block .product-title,
      #wsite-content .product-description,
      #wsite-content .wsite-form-field label,
      #wsite-content .wsite-form-field label,
      .blog-sidebar div.paragraph,
      .blog-sidebar p,
      .blog-sidebar .wsite-form-field label,
      .blog-sidebar .wsite-form-field label {}

      #wsite-content div.paragraph,
      #wsite-content p,
      #wsite-content .product-block .product-title,
      #wsite-content .product-description,
      #wsite-content .wsite-form-field label,
      #wsite-content .wsite-form-field label,
      .blog-sidebar div.paragraph,
      .blog-sidebar p,
      .blog-sidebar .wsite-form-field label,
      .blog-sidebar .wsite-form-field label {}

      .wsite-elements.wsite-footer div.paragraph,
      .wsite-elements.wsite-footer p,
      .wsite-elements.wsite-footer .product-block .product-title,
      .wsite-elements.wsite-footer .product-description,
      .wsite-elements.wsite-footer .wsite-form-field label,
      .wsite-elements.wsite-footer .wsite-form-field label {}

      .wsite-elements.wsite-not-footer:not(.wsite-header-elements) h2,
      .wsite-elements.wsite-not-footer:not(.wsite-header-elements) .product-long .product-title,
      .wsite-elements.wsite-not-footer:not(.wsite-header-elements) .product-large .product-title,
      .wsite-elements.wsite-not-footer:not(.wsite-header-elements) .product-small .product-title,
      #wsite-content h2,
      #wsite-content .product-long .product-title,
      #wsite-content .product-large .product-title,
      #wsite-content .product-small .product-title,
      .blog-sidebar h2 {}

      #wsite-content h2,
      #wsite-content .product-long .product-title,
      #wsite-content .product-large .product-title,
      #wsite-content .product-small .product-title,
      .blog-sidebar h2 {}

      .wsite-elements.wsite-footer h2,
      .wsite-elements.wsite-footer .product-long .product-title,
      .wsite-elements.wsite-footer .product-large .product-title,
      .wsite-elements.wsite-footer .product-small .product-title {}

      #wsite-title {}

      .wsite-menu-default a {}

      .wsite-menu a {}

      .wsite-image div,
      .wsite-caption {}

      .galleryCaptionInnerText {}

      .fancybox-title {}

      .wslide-caption-text {}

      .wsite-phone {}

      .wsite-headline,
      .wsite-header-section .wsite-content-title {}

      .wsite-headline-paragraph,
      .wsite-header-section .paragraph {}

      .wsite-button-inner {}

      .wsite-not-footer blockquote {}

      .wsite-footer blockquote {}

      .blog-header h2 a {}

      #wsite-content h2.wsite-product-title {}

      .wsite-product .wsite-product-price a {}
    }
  </style>
  <script>
    var STATIC_BASE = '//cdn1.editmysite.com/';
    var ASSETS_BASE = '//cdn2.editmysite.com/';
    var STYLE_PREFIX = 'wsite';
  </script>
  <script src='https://ajax.googleapis.com/ajax/libs/jquery/1.8.3/jquery.min.js'></script>
  <script type="text/javascript" src="http://cdn2.editmysite.com/js/lang/en/stl.js?buildTime=1234&"></script>
  <script src="http://cdn2.editmysite.com/js/site/main.js?buildTime=1234"></script>
  <script type="text/javascript">_W.configDomain = "www.weebly.com";</script>
  <script>_W.relinquish && _W.relinquish()</script>
  <script type="text/javascript" src="http://cdn2.editmysite.com/js/lang/en/stl.js?buildTime=1234&"></script>
  <script> _W.themePlugins = [];</script>
  <script type="text/javascript"> _W.recaptchaUrl = "https://www.google.com/recaptcha/api.js"; </script>
  <script type="text/javascript"> window._W = window._W || {}; _W.showV2Footer = 0; </script>
  <script type="text/javascript">< !--
      var IS_ARCHIVE = 1;

    function initFlyouts() {
      initPublishedFlyoutMenus(
        [{ "id": "357357369650560692", "title": "Home", "url": "index.html", "target": "", "nav_menu": false, "nonclickable": false }, { "id": "498844820849822601", "title": "About", "url": "about.html", "target": "", "nav_menu": false, "nonclickable": false }, { "id": "298570032218330071", "title": "Competition tasks", "url": "competition-tasks.html", "target": "", "nav_menu": false, "nonclickable": false }, { "id": "137293036424937964", "title": "AGENDA", "url": "agenda.html", "target": "", "nav_menu": false, "nonclickable": false }, { "id": "570574741552950396", "title": "Contact", "url": "contact.html", "target": "", "nav_menu": false, "nonclickable": false }, { "id": "241401416158911963", "title": "Registration (competition)", "url": "registration-competition.html", "target": "", "nav_menu": false, "nonclickable": false }, { "id": "417504476769125933", "title": "Registration (workshop)", "url": "registration-workshop.html", "target": "", "nav_menu": false, "nonclickable": false }],
        "298570032218330071",
        '',
        'active',
        false,
        { "navigation\/item": "<li {{#id}}id=\"{{id}}\"{{\/id}} class=\"wsite-menu-item-wrap\">\n\t<a\n\t\t{{^nonclickable}}\n\t\t\t{{^nav_menu}}\n\t\t\t\thref=\"{{url}}\"\n\t\t\t{{\/nav_menu}}\n\t\t{{\/nonclickable}}\n\t\t{{#target}}\n\t\t\ttarget=\"{{target}}\"\n\t\t{{\/target}}\n\t\t{{#membership_required}}\n\t\t\tdata-membership-required=\"{{.}}\"\n\t\t{{\/membership_required}}\n\t\tclass=\"wsite-menu-item\"\n\t\t>\n\t\t{{{title_html}}}\n\t<\/a>\n\t{{#has_children}}{{> navigation\/flyout\/list}}{{\/has_children}}\n<\/li>\n", "navigation\/flyout\/list": "<div class=\"wsite-menu-wrap\" style=\"display:none\">\n\t<ul class=\"wsite-menu\">\n\t\t{{#children}}{{> navigation\/flyout\/item}}{{\/children}}\n\t<\/ul>\n<\/div>\n", "navigation\/flyout\/item": "<li {{#id}}id=\"{{id}}\"{{\/id}}\n\tclass=\"wsite-menu-subitem-wrap {{#is_current}}wsite-nav-current{{\/is_current}}\"\n\t>\n\t<a\n\t\t{{^nonclickable}}\n\t\t\t{{^nav_menu}}\n\t\t\t\thref=\"{{url}}\"\n\t\t\t{{\/nav_menu}}\n\t\t{{\/nonclickable}}\n\t\t{{#target}}\n\t\t\ttarget=\"{{target}}\"\n\t\t{{\/target}}\n\t\tclass=\"wsite-menu-subitem\"\n\t\t>\n\t\t<span class=\"wsite-menu-title\">\n\t\t\t{{{title_html}}}\n\t\t<\/span>{{#has_children}}<span class=\"wsite-menu-arrow\">&gt;<\/span>{{\/has_children}}\n\t<\/a>\n\t{{#has_children}}{{> navigation\/flyout\/list}}{{\/has_children}}\n<\/li>\n" },
        {}
      )
    }
      //-->
  </script>
</head>

<body class="wsite-theme-light short-header-page  wsite-page-competition-tasks">
  <div id="header-wrap">
    <div id="page">
      <div id="header-container">
        <table id="header">
          <tr>
            <td id="logo"><span class="wsite-logo">
                <a href="">
                  <span id="wsite-title">IDASH PRIVACY & SECURITY WORKSHOP 2022 - secure genome analysis
                    competition</span>
                </a>
              </span>
            </td>
            <td id="header-right">
              <table>
                <tr>
                  <td class="phone-number"><span class="wsite-text wsite-phone">
                      *NHGRI R13HG009072
                    </span>
                  </td>
                  <td class="social">
                    <div style="text-align:left;">
                      <div style="height:0px;overflow:hidden"></div>
                      <span class="wsite-social wsite-social-default"></span>
                      <div style="height:0px;overflow:hidden"></div>
                    </div>
                  </td>
                </tr>
              </table>
              <div class="search"></div>
            </td>
          </tr>
        </table>
        <div id="topnav">
          <ul class="wsite-menu-default">
            <li id="pg357357369650560692" class="wsite-menu-item-wrap">
              <a href="index.html" class="wsite-menu-item">
                Home
              </a>
            </li>
            <li id="pg498844820849822601" class="wsite-menu-item-wrap">
              <a href="about.html" class="wsite-menu-item">
                About
              </a>
            </li>
            <li id="active" class="wsite-menu-item-wrap">
              <a href="competition-tasks.html" class="wsite-menu-item">
                Competition tasks
              </a>
            </li>
            <li id="pg137293036424937964" class="wsite-menu-item-wrap">
              <a href="agenda.html" class="wsite-menu-item">
                AGENDA
              </a>
            </li>
            <li id="pg570574741552950396" class="wsite-menu-item-wrap">
              <a href="contact.html" class="wsite-menu-item">
                Contact
              </a>
            </li>
            <li id="pg241401416158911963" class="wsite-menu-item-wrap">
              <a href="registration-competition.html" class="wsite-menu-item">
                Registration (competition)
              </a>
            </li>
            <li id="pg417504476769125933" class="wsite-menu-item-wrap">
              <a href="registration-workshop.html" class="wsite-menu-item">
                Registration (workshop)
              </a>
            </li>
          </ul>
          <div style="clear:both"></div>
        </div>
      </div>
    </div>
  </div>
  <div id="banner-wrap">
    <div id="container">
      <div id="banner-bot">
        <div id="banner">
          <div id="banner-mid">
            <div id="banner-outer">
              <div class="wsite-header"></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div id="main-wrap">
    <div id="page">
      <div id="main">
        <div id="content">
          <div id="wsite-content" class="wsite-elements wsite-not-footer">


            <h2 class="wsite-content-title" style="text-align:left;"><font size="7">Four competition tasks</font></h2> <p><font size="3"> </p>
  
<style type="text/css">
<!--
 .tab { margin-left: 20px; }
-->
</style>  
            
<div class="paragraph" style="text-align:left;">
    <span>
      <font color="#626262"><span style="font-weight:700"> <font size="6"> Task 1: Blockchain-based Recording of Biomedical and Genomic Compliance Training Certificates
      </span></font> 
<br/>
(organized by Tsung-Ting Kuo and Lucila Ohno-Machado)

<br/>
<br/>            

<span><span style="font-weight:700"><font color="#626262">Background:</font></span><span>&nbsp;</span><font color="#818181">  Training certificates help ensure that researchers accessing sensitive protected data have received sufficient training to hand it in compliance with all necessary regulations. Many such certificates are still in the form of PDF files (e.g., CITI [1]). Using the information contained in such training certificates can be critical for education, research, and other practical purposes. Individuals have access to their certificates and provide them to requesting institutions. Centralized solutions are not ideal since certificates are issued by various authorities and individuals want to have control over them. Additionally, centralized solutions such as databases incur the risk of serving as a single-point-of-failure.[2] The purpose of this task is to benchmark the use of blockchain, a decentralized technology, for storing and retrieving certificates.

<br/>
<br/>

<span><span style="font-weight:700"><font color="#626262">Goal:</font></span><span>&nbsp;</span><font color="#818181">  The goal of this track is to develop smart contracts on a blockchain network to record and query human subjects’ training certificates or similar documents. Specifically, given a set of training certificate PDF files, which are truncated into multiple sections of binary data (called “chunks,” and  their metadata (participant name, certificate name, date of completion, etc.), contestants will design time-efficient data structures and algorithms based on Ethereum Solidity to store and retrieve such data.

<br/>
<br/>


<span><span style="font-weight:700"><font color="#626262">Challenge: </font></span><span>&nbsp;</span><font color="#818181"> All data (including chunks and their metadata) and intermediary data (e.g., index or cache of the original data) must be saved on-chain (i.e., no off-chain data storage is allowed) via smart contracts. We will provide the skeleton of the smart contracts and scripts. Note that the implementation of a smart contract is required to allow the insertion of one chunk and its accompanying metadata. Each participant can determine how each insertion is represented and stored in the smart contract.  The two query functions should return (1) all metadata fulfilling a search criterion, and (2) the latest PDF file fulfilling a search criterion. Participants should not use any third party libraries. There will be n nodes (e.g., n = 5) in the blockchain network. Users should be able to query the data from any of the nodes. Participants can implement any algorithm to store, retrieve and present the data correctly and efficiently.

<br/>
<br/>



<span><span style="font-weight:700"><font color="#626262">Evaluation: </font></span><span>&nbsp;</span><font color="#818181"> The data sharing system will need to demonstrate satisfactory performance (i.e., accurate query results) on a test dataset, which will be different from the training set provided online. The efficiency of each solution will be evaluated as follows. First, we will insert the certificate chunks and metadata lines from the nodes for a certain time (e.g., 30 minutes) in the blockchain network. Then, we will query all nodes to verify that the insertions were done correctly. Finally, we will compute the average number of records inserted per second (<b style='color:red;'>33.33%</b>), the average time to run the first query (<b style='color:red;'>33.33%</b>), and the average time to run the second query (<b style='color:red;'>33.33%</b>) as our final evaluation metric. We plan to evaluate the results using Go-Ethereum 1.10.7, Solidity 0.8.12 (with ABI Encoder v2 enabled), and Ubuntu 18.04.5. Submissions will not be evaluated using any other platforms/versions for fairness. No modification of the underlying Go-Ethereum source code is allowed. The submission should include one file of Solidity source code.

<br/>
<br/>


<span><span style="font-weight:700"><font color="#626262">License: </font></span><span>&nbsp;</span><font color="#818181"> This track requires every participating team to share their code and/or binaries under the BSD 3-Clause License Open Source license. The track co-organizers will upload all submitted code and/or binaries to a GitHub repository under the BSD 3-Clause License right after the competition results are announced. By submitting the code and/or binaries, the participants automatically consent to allow the track co-organizers to release the code and/or binaries under the BSD 3-Clause License.

<br/>



     <br /><span><span style="font-weight:700"><font color="#626262">Data and Code Skeleton: <a href="https://ucsdcloud-my.sharepoint.com/:u:/g/personal/tskuo_ucsd_edu/EeWZByRBRxpJhtLLwHlhCVMBfFtWRq-ZMvKtGVAT5tRLzw?e=pHTuFP" target="_blank">link</a></span></span> 

    <br />

         <br /><span><span style="font-weight:700"><font color="#626262">FAQ: <a href="https://docs.google.com/document/d/1ke7ZU10lpm-4btoqj9Gl6BMGO5TBEG0Vl02d_DmEVAQ/edit?usp=sharing" target="_blank">link</a></span></span> 

    <br />

<br/>
<span><span style="font-weight:700"><font color="#626262">Reference: </font></span><span>&nbsp;</span><font color="#818181"> [1] Collaborative Institutional Training Initiative (CITI), https://about.citiprogram.org/en/homepage/

<br/>

[2] Tellew J, Kuo T-T. CertificateChain: decentralized healthcare training certificate management system using blockchain and smart contracts. JAMIA Open. 2022;5(1). doi: 10.1093/jamiaopen/ooac019. PubMed PMID: 35571362.

  <br/>











<style type="text/css">
<!--
 .tab { margin-left: 20px; }
-->
</style>  
            
<div class="paragraph" style="text-align:left;">
    <span>
      <font color="#626262"><span style="font-weight:700"> <font size="6"> Task 2: Secure Model Evaluation on Homomorphically Encrypted Genotype Data 
      </span></font> 
<br/>
(organized by Miran Kim, Arif Harmanci, Xiaoqian Jiang)

<br/>
<br/>            

<span><span style="font-weight:700"><font color="#626262">Background:</font></span><span>&nbsp;</span><font color="#818181">  Prediction of phenotypes from high throughput genomic data is an important problem in genomics. Especially with the emergence of large genotype-phenotype databases, the models for phenotype prediction can provide an accurate estimation of phenotype information and disease risk. 3rd parties such as genealogy companies routinely use genetic data to estimate and report phenotype information to customers. Because of the sensitive nature of genetic and phenotypic data, the confidentiality of individuals, patients, and database participants is becoming a major challenge. 

<br/>
<br/>

<span><span style="font-weight:700"><font color="#626262">Challenge:</font></span><span>&nbsp;</span><font color="#818181">  We are challenging participants to build a secure outsourcing protocol for phenotype prediction whereby the genotype and model parameters are both protected. The participants will be asked to develop the prediction models (training in plaintext) and the secure model evaluation protocol (test in ciphertext). We expect the model to be trained in plaintext using the challenge dataset that contains genotypes and multiple phenotypes. The phenotypes will be based on simulations by organizers generated from complex functions of genotypes (due to restricted access requirements of existing datasets, we will not be able to provide real data for this challenge). The encrypted trained model will be used for the evaluation of homomorphically encrypted data (assuming it can be hosted on untrusted servers) for performance measurement. The main challenge stems from the optimization of model accuracy and evaluation performance. For example, more complex models can perform better predictions, but they may incur a high computational cost, which might render the computation practically infeasible. 
<br/>
<br/>


<span><span style="font-weight:700"><font color="#626262">Experimental setting: </font></span><span>&nbsp;</span><font color="#818181"> In this scenario, there are 3 parties: Client, Modeler, and Evaluator. The client wants to use their sensitive genotype data to perform phenotype prediction using the Modeler's models. Modeler builds the phenotype prediction models that take genotypes as input. Models contain sensitive information and cannot be shared in plain form. Therefore, Modeler releases their models only in encrypted form. The evaluator performs the outsourcing of model evaluation by taking encrypted genotype and encrypted model parameters. The challenge involves building the models (for Modeler) and the secure evaluation of the models on encrypted genotype data (for Evaluator). As described above, the models and genotype data are sensitive and must be encrypted before they are sent to the Evaluator. Submission site: https://uthtmc.az1.qualtrics.com/jfe/form/SV_8rlYH102Sl4e7qu



<br/>
<br/>


<span><span style="font-weight:700"><font color="#626262">Requirement: </font></span><span>&nbsp;</span><font color="#818181">  This year, we do not allow any preprocessing operations on the genotype data before encryption, i.e., there can be no plaintext preprocessing of the genotype data before it is encrypted as these preprocessing operations can leak information about the prediction models. The participants can use any encryption scheme they wish as long as 128-bit security is satisfied as per the homomorphic encryption white paper [1].

<br/>
<br/>


<span><span style="font-weight:700"><font color="#626262">Evaluation Criteria: </font></span><span>&nbsp;</span><font color="#818181"> All submissions will be tested on a hold-out dataset of 198 genomes in an isolated environment in terms of performance. Accuracy and time/memory requirements will be used for benchmarking and ranking the submissions. The submissions will be required to finish all protocol in less than 30 minutes. Please refer to README file for more details."
<br/>
<br/>


     <br /><span><span style="font-weight:700"><font color="#626262">Data and Code Skeleton: <a href="https://drive.google.com/file/d/1LdUOBKH_m44VSk7JlLGYqasXarjdVRFl/view?usp=sharing" target="_blank">link</a></span></span> 

    <br />

         <br /><span><span style="font-weight:700"><font color="#626262">FAQ: <a href="https://docs.google.com/document/d/1X3SS6zQiUrC6hW8r3gAOA2a085oPAUYQrXJE-ZRrsXw/edit" target="_blank">link</a></span></span> 

    <br />

<br/>

<span><span style="font-weight:700"><font color="#626262">License: </font></span><span>&nbsp;</span><font color="#818181"> This track requires every participating team to share their code and/or binaries under the BSD 3-Clause License Open Source license. The track co-organizers will upload all submitted code and/or binaries to a GitHub repository under the BSD 3-Clause License right after the competition results are announced. By submitting the code and/or binaries, the participants automatically consent to allow the track co-organizers to release the code and/or binaries under the BSD 3-Clause License.

<br/>
<br/>


<span><span style="font-weight:700"><font color="#626262">Reference: </font></span><span>&nbsp;</span><font color="#818181"> [1] Albrecht et al., Homomorphic Encryption Standard, https://eprint.iacr.org/2019/939.pdf

<br/>













<style type="text/css">
<!--
 .tab { margin-left: 20px; }
-->
</style>  
            
<div class="paragraph" style="text-align:left;">
    <span>
      <font color="#626262"><span style="font-weight:700"> <font size="6"> Task 3: Confidential Computing for clustering single-cell transcriptomics data
      </span></font> 
<br/>
(organized by Haixu Tang and Xiaofeng Wang)

<br/>
<br/>            

<span><span style="font-weight:700"><font color="#626262">Background:</font></span><span>&nbsp;</span><font color="#818181">  In the past few years, single-cell RNA-seq technologies have advanced rapidly. Un-supervised learning methods such as dimension reduction and clustering algorithms are now widely used to discover different types or subtypes of cells in hundreds to thousands of single cells e.g., from tumor or normal tissues, based on their gene expression profiles. Previous research has shown gene expression patterns may reveal identifiable information about the donor of the tissues; therefore, proper protection of such data should be in place when these data are re-analyzed (e.g., in a meta-study) by an untrusted user and/or on a public computing environment (e.g., on a commercial cloud). Trusted Executive Environment (TEE, e.g., Intel's SGX) provides an ideal infrastructure for hosting such privacy-preserving analyses of single-cell transcriptomics data by a data user, in which 1) only the computing task (i.e., clustering of single-cell gene expression profiles in this case) approved by the data owner of the input data is allowed be performed on the data; and 2) the data user does not see the content of input data, which are encrypted in a way that only the client and the TEE can decrypt. In 2020, we organized a challenge for efficient implementation of a single cell clustering algorithm within a single enclave using Intel's SGX technique. The results showed that the secure algorithm is efficient for clustering thousands of cells, but is not scalable well to tens of thousands of cells. As the current single cell RNA-seq analyses routinely involve tens to hundreds of thousands of cells, scalable clustering algorithms implemented using multiple enclaves (i.e., allowing for parallel computing) are required. The purpose of this task is to test the scalability of a parallel unsupervised clustering algorithm  using multiple enclaves in SGX when applied to massive single-cell RNA-seq data.
<br/>
<br/>

<span><span style="font-weight:700"><font color="#626262">Challenge:</font></span><span>&nbsp;</span><font color="#818181"> We challenge participating teams to implement a  parallel clustering algorithm (such as Para-DPMM [1], but can be a different clustering algorithm)  on the Intel SGX platform, so that  the algorithm can operate on multiple enclaves using up to 4 threads in each enclave. Note that here we attempt to simulate an elastic computing environment in a public cloud service, on which a single CPU package may be used by multiple jobs from different users, and thus only limited resources (i.e., 4 threads) are allocated to each enclave when it is initiated. The implementation should protect both the input data: that is, any input, intermediate and output data should be encrypted outside the enclave (including communicated between enclaves). However, we do not consider side channel leaks in this task.

<br/>
<br/>


<span><span style="font-weight:700"><font color="#626262">Experimental setting: </font></span><span>&nbsp;</span><font color="#818181"> We will provide a testing dataset along with the reference implementation of Para-DPMM clustering algorithm without using SGX. Each team is challenged to implement a parallel clustering algorithm (not necessarily DPMM) using SGX. For this purpose, the team is allowed to develop its own parallel algorithms as well as the methods to exchange data and intermediate results (they should be encrypted under the security requirement) between enclaves as long as the overall clustering accuracy is largely preserved and the data are fully protected (except for side channel leaks). The testing datasets of different sizes (number of cells and genes) will be used to evaluate the scalability of the submitted solutions. The solution may utilize the computational resources outside the enclave, including the CPU, memory and hard disk, as long as all the data and the model are fully protected (encrypted at least 128-bit security level). The submitted solution cannot involve any additional party. Pre-computing time will be measured as part of the performance overhead.

<br/>
<br/>


<span><span style="font-weight:700"><font color="#626262">Requirement: </font></span><span>&nbsp;</span><font color="#818181"> Each participating team should submit their implementation together with source code. We will provide remote access to an SGX system for the team to install their implementation for evaluation. The teams are responsible for the compatibility of their implementation and the system. The evaluation will be done using un-released testing data.

<br/>
<br/>


<span><span style="font-weight:700"><font color="#626262">Evaluation Criteria: </font></span><span>&nbsp;</span><font color="#818181"> All submissions should meet security requirements (at least 128-bit security level). The secure algorithms will then be evaluated based on the quality of the clustering results (measured by Adjusted Rand Index or ARI). We will also compare different submissions' performance (running time) when their accuracy comes close.


<br/>

    <br /><span><span style="font-weight:700"><font color="#626262">Test Platform Information</font></span><span>&nbsp;</span><font color="#818181"><br> OS: Ubuntu 20.04

              <br> Kernel: 5.15.0-46-generic
              <br> CPU: Intel(R) Xeon(R) Platinum 8352Y (2-socket)

              <br> Memory: 256 GB
              <br> EPC Size: 128 GB


    <br />

     <br /><span><span style="font-weight:700"><font color="#626262">Data and Code Skeleton(password: iDash2022): <a href="https://indiana-my.sharepoint.com/:f:/g/personal/zhu11_iu_edu/Ek3KFg3_saxFlYETlLo0XLMBgAfE51EAWJI3EWUyUsi6MQ?e=dDkk3b" target="_blank">link</a></span></span> 

    <br />

      <br /><span><span style="font-weight:700"><font color="#626262">FAQ: <a href="https://docs.google.com/document/d/1uQveDd_5RDLuRrNwH08GUxTb8zwUtORZSwJlcC2Q9FQ/edit?usp=sharing" target="_blank">link</a></span></span> 

    <br />


<br/>




<span><span style="font-weight:700"><font color="#626262">Reference: </font></span><span>&nbsp;</span><font color="#818181"> [1] Duan, T., Pinto, J.P. and Xie, X., 2019. Parallel clustering of single cell transcriptomic data with split-merge sampling on Dirichlet process mixtures. Bioinformatics, 35(6), pp.953-961.

<br/>













<style type="text/css">
<!--
 .tab { margin-left: 20px; }
-->
</style>  
            
<div class="paragraph" style="text-align:left;">
    <span>
      <font color="#626262"><span style="font-weight:700"> <font size="6"> Task 4:  Secure Record Linkage 
      </span></font> 
<br/>
(organized by Haixu Tang and Xiaofeng Wang in collaboration with Ruiyu@Meta)


<br/>
<br/>            

<span><span style="font-weight:700"><font color="#626262">Background:</font></span><span>&nbsp;</span><font color="#818181">   Record linkage, which aims to link the records from two or more distinct sources, has been commonly adopted in biomedical informatics to link electronic health records (EHRs) of the same patient in different medical institutions, e.g., for building the longitudinal health profile of the patient or for de-duplicating individual records in multiple databases, etc. Due to the concerns of patients’ confidentiality and/or the institutional data sharing policy, the patient records, in particular the identity keys (e.g, the patients’ name, SSN, address, etc), cannot be directly shared with the other party. Therefore, a secure algorithm is needed to achieve record linkage while the protected keys are only shared in their encrypted form.
In a typical scenario of secure record linkage, two parties each hold a private data set of patients’ records, and wish to identify the patients shared by both data sets based on the matched identity keys (e.g. first/last name, email, phone number, SSN, address/zip code, birth date, gender, etc). Notably, in practice, the record linkage is non-trivial because 1) keys may be missing, while the others can be inaccurate in both datasets, e.g., the address may be changed, the name may contain typos etc; and 2) one dataset may contain multiple records from the same or very similar (e.g., from the same family) patients. Here, the task is to implement a secure record linkage algorithm for two parties, each holding a private dataset of patients’ records, to identify the shared records using multiple, possibly non-unique, keys without sharing each party’s dataset with the other party in plaintext. We expect the algorithm is scalable to the datasets consisting of millions of records, and does not involve a trusted third party in the computation.

<br/>
<br/>

<span><span style="font-weight:700"><font color="#626262">Challenge:</font></span><span>&nbsp;</span><font color="#818181"> We challenge the participating teams to implement a secure two-party computing algorithm that uses a scoring algorithm based on multiple identity keys to identify matched records. The scoring function should be designed to reflect the different weights of different keys: some keys have deterministic weights if they are not missing (e.g., SSN), while the other keys (e.g., birth dates, email address, etc) have lower weights to depending on their potential errors. The error rate and the probability of missing value for each key will be given in advance. The algorithm will output a list of matched records (IDs) ranked by the scores. The participating teams may consider (but not required) to use the MPC framework developed by Meta (https://github.com/facebookresearch/fbpcf) to implement their solutions.
<br/>
Input format: A database of patient records (a set of keys) for each party. Each record in either database is assigned to a unique ID, which should only be used  in the output. Some of the records have missing data. In some non-rare cases, multiple records share some identity keys (e.g. same first/last name, same address). 
<br/>
Output format: The two parties get a secret-shared list of IDs representing the matched records.

<br/>
<br/>


<span><span style="font-weight:700"><font color="#626262">Security requirement: </font></span><span>&nbsp;</span><font color="#818181"> The two semi-honest parties should learn no extra information except:
<br/>
1.Number of records in peer’s database
<br/>
2.Number of matched records
<br/>
3.Number of missing values for each identity key


<br/>
<br/>




<span><span style="font-weight:700"><font color="#626262">Evaluation Criteria: </font></span><span>&nbsp;</span><font color="#818181">  Each team should submit a single solution. If you plan to submit multiple solutions that use substantially different approaches, please contact the organizers to get approval. All submitted solutions should fulfill the security requirement; otherwise, the submission will be disqualified. The qualified solutions will be ranked based on their accuracy (i.e., outputting the correct list of matched records). For the solutions with the same or very close (i.e., within 1% difference), we will evaluate the speed of the algorithm. For detail of the data, please see it in the FAQ. The guideline for submitting your solutions will be available at the website later(Please send an email to zhu11@iu.edu if your team wants to submit your result. Your email should include your team name. We will create one drive folder for your team, and you should submit your solution to this one drive folder. Note that each team can only submit one solution, so please send only one email to us if your team wants to submit your result.) We will run the program no more than 24 hours. You may use up to 32G memory and the CPU is quad. 

<br/>


     <br /><span><span style="font-weight:700"><font color="#626262">Data and Code Skeleton(Updated at Aug 11, password:iDash2022): <a href="https://indiana-my.sharepoint.com/:f:/g/personal/zhu11_iu_edu/EkFJ17EHX59LsO1Ekfc0TPkBi7CipeIHccd4wjb0CRhjzQ?e=qlt6aH" target="_blank">link</a></span></span> 

    <br />

         <br /><span><span style="font-weight:700"><font color="#626262">FAQ: <a href="https://www.notion.so/huthvincent/All-you-need-to-know-in-the-Track-4-5f1948397ca246d197c9555a1663f80f" target="_blank">link</a></span></span> 

    <br />


<br/>

<span><span style="font-weight:700"><font color="#626262">Submission: </font></span><span>&nbsp;</span><font color="#818181">  Each participating should submit the solution in two containers based on official Ubuntu docker image, of which each can be run on an independent server running Ubuntu. The containers should set up a communication channel to exchange the (protected) input and other necessary data. Each team is expected to also submit the source code of their solution. The source code will only be used for evaluation purpose (i.e., checking the security compliance) by the organizers.


<br/>
<br/>









              <div style="height: 20px; overflow: hidden; width: 100%;"></div>
            </div>
          </div>
        </div>
      </div>
    </div>
</body>

</html>